variables.tf

```diff
@@ -2,10 +2,174 @@
 # Input Variables
 ##############################################################################
 
-#variable "myvariable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = ""
-#}
+variable "ibmcloud_api_key" {
+  description = "APIkey that's associated with the account to use, set via environment variable TF_VAR_ibmcloud_api_key"
+  type        = string
+  sensitive   = true
+}
+
+# Resource Group Variables
+variable "resource_group_id" {
+  type        = string
+  description = "The Id of an existing IBM Cloud resource group where the cluster will be grouped."
+}
+
+variable "region" {
+  type        = string
+  description = "The IBM Cloud region where the cluster will be provisioned."
+}
+
+# Cluster Variables
+variable "tags" {
+  type        = list(string)
+  description = "Metadata labels describing this cluster deployment, i.e. test"
+  default     = []
+}
+
+variable "cluster_name" {
+  type        = string
+  description = "The name that will be assigned to the provisioned cluster"
+}
+
+variable "vpc_subnets" {
+  type = map(list(object({
+    id         = string
+    zone       = string
+    cidr_block = string
+  })))
+  description = "Metadata that describes the VPC's subnets. Obtain this information from the VPC where this cluster will be created"
+}
+
+variable "worker_pools" {
+  type = list(object({
+    subnet_prefix     = string
+    pool_name         = string
+    machine_type      = string
+    workers_per_zone  = number
+    resource_group_id = optional(string)
+    labels            = optional(map(string))
+  }))
+  default = [
+    {
+      subnet_prefix    = "zone-1"
+      pool_name        = "default" # ibm_container_vpc_cluster automatically names default pool "default" (See https://github.com/IBM-Cloud/terraform-provider-ibm/issues/2849)
+      machine_type     = "bx2.4x16"
+      workers_per_zone = 2
+      labels           = {}
+    },
+    {
+      subnet_prefix    = "zone-2"
+      pool_name        = "zone-2"
+      machine_type     = "bx2.4x16"
+      workers_per_zone = 2
+      labels           = { "dedicated" : "zone-2" }
```

Comment:
I would leave those labels and the associated worker pools taints out of the default as this is confusing for the public version which allows any type of topology. Perhaps this could be in one of the example / a new example showing how to set worker pool taints.

Reply:
Also add a note that at least one of the "pool_name" must be named "default" in the documentation

Reply:
Have mentioned in the root documentation as a note about the default pool name.

Reply:
Removed the taints and labels from the default values in the example but not created a new example for setting worker pool taints. Can we track it separately like few others?

Reply:
@imprateeksh - Yep that's fine. thanks.

Reply:
@imprateeksh - The default for labels and worker pools taints is still set in the root module. Did you mean to remove them in this PR (missing commit?)

Reply:
@vburckhardt - Sure. I will merge these changes in another PR.